docs/predictive-analytics/next-best-offer/index.md

```diff
@@ -0,0 +1,208 @@
+---
+seo:
+  title: Next Best Offer for Banking
+  description: This recipe demonstrates how to use ksqlDB to present relevant offers to banking customers.
+---
+
+# Next Best Offer for Banking
+
+Customers today are faced with never ending marketing messages from a variety of sources.  Often these messages are generic and don't have any consideration for the individual needs of the customer.  This one-size-fits-all approach leads to poor conversion rates.  A better approach is to tailor offerings that take into consideration the interests of the customer based on previous purchases or behavior.   This recipe demonstrates how to take existing customer information and provide a "Next Best Offer" to encourage sales and retain customers.
+
+## Step-by-step
+
+### Set up your environment
+
+Provision a Kafka cluster in [Confluent Cloud](https://www.confluent.io/confluent-cloud/tryfree/?utm_source=github&utm_medium=ksqldb_recipes&utm_campaign=next_best_offer).
+
+
+--8<-- "docs/shared/ccloud_setup.md"
+
+### Read the data in
+
+--8<-- "docs/shared/connect.md"
+
+```json
+--8<-- "docs/predictive-analytics/next-best-offer/source.json"
+```
+
+--8<-- "docs/shared/manual_insert.md"
+
+### Run the stream processing application
+
+This application will perform a series of joins between event streams and tables to calculate the next best offer for a banking customer based on their activity which should yield higher customer activity and satisfaction.
+
+--8<-- "docs/shared/ksqlb_processing_intro.md"
+
+```sql
+--8<-- "docs/predictive-analytics/next-best-offer/process.sql"
+```
+
+--8<-- "docs/shared/manual_cue.md"
+
+```sql
+--8<-- "docs/predictive-analytics/next-best-offer/manual.sql"
+```
+
+
+### Cleanup
+
+--8<-- "docs/shared/cleanup.md"
+
+## Explanation
+
+### Creating an event stream
+
+To get started, you'll first need to create a stream that contains the customer activity:
+
+```sql
+CREATE STREAM customer_activity_stream (
+    ACTIVITY_ID INTEGER,
+    IP_ADDRESS STRING,
+    CUSTOMER_ID INTEGER KEY,
+    ACTIVITY_TYPE STRING,
+    PROPENSITY_TO_BUY DOUBLE
+   ) WITH (
+    KAFKA_TOPIC = 'CUSTOMER_ACTIVITY_STREAM',
+    VALUE_FORMAT = 'JSON',
+    PARTITIONS = 6
+);
+```
+
+Note that the `CUSTOMER_ID` field is the key in the stream's key-value pairs. You'll see why this is important in the next section.
+
+In a production setting you'll populate the stream's underlying topic either with `KafkaProducer` application or from an external system using a [managed connector on Confluent Cloud](https://docs.confluent.io/cloud/current/connectors/index.html).  In case you don't have a connector set up, you can manually insert records into the stream with [INSERT VALUES](https://docs.ksqldb.io/en/latest/developer-guide/ksqldb-reference/insert-values/#insert-values) statements:
+
+
+```sql
+INSERT INTO customer_activity_stream (activity_id, ip_address, customer_id, activity_type, propensity_to_buy) VALUES (1,'121.219.110.170',1,'branch_visit',0.4);
```

Comment:
If we change the column order above, we would need to update these statements, too.